src/climate_learn/trainer.py

```diff
@@ -0,0 +1,71 @@
+# Standard library
+import logging
+import sys
+from warnings import warn
+
+# Third party
+import torch
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import (
+    ModelCheckpoint,
+    RichModelSummary,
+    RichProgressBar,
+    EarlyStopping,
+)
+
+logging.getLogger("pytorch_lightning").setLevel(logging.ERROR)
+
+
+class Trainer(pl.Trainer):
+    """Wrapper for Lightning's trainer."""
+
+    def __init__(
+        self,
+        early_stopping=None,
+        patience=0,
+        summary_depth=-1,
+        seed=0,
+        **kwargs
+    ):
+        pl.seed_everything(seed)
+        if "accelerator" not in kwargs and torch.cuda.is_available():
+            print(
+                "GPU detected but 'accelerator' not specified..."
+                " setting 'accelerator' = \"gpu\""
+            )
+            kwargs["accelerator"] = "gpu"
```

Comment:
Good point. I'll change this so that if `accelerator` is not specified, then even if CUDA is available, the trainer will not use GPU acceleration. Only if the user explicitly requests GPU acceleration will the trainer do so.